serif_swing/src/main/kotlin/serif_swing/App.kt

```diff
@@ -939,13 +964,55 @@ class SwingChatRoom(val transition: (MatrixState, Boolean) -> Unit, val panel: J
     val message_field = SmoothTextField(20)
     val pinned_events_btn = SmoothButton("Pinned Events")
     val pinned_action_popup = JPopupMenu()
+    val mentions_popup = JPopupMenu()
     val msg_context_label = SmoothLabel("Reply")
     val msg_context_panel = JPanel()
 
     var replied_event_id = ""
     var reacted_event_id = ""
     var edited_event_id = ""
+    var formatted = false
```

Comment:
If we do go with the getUnformattedBody changes thing way, we can remove this! Which I'm getting more on board with